lib/widgets/message_list.dart

```diff
@@ -643,6 +651,35 @@ class _UnreadMarker extends StatelessWidget {
   }
 }
 
+class MarkAsReadAnimation extends StatelessWidget {
+  const MarkAsReadAnimation({
+    super.key,
+    required this.visible,
+    required this.loading,
+    required this.child});
+
+  final bool visible;
+  final bool loading;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    final opacity = loading ? 0.5 : visible ? 1.0 : 0.0;
+    final scale = loading ? 0.95 : 1.0;
```

Comment:
After loading is finished the button will scale up instead of scaling down, unlike mentioned [here](https://github.com/zulip/zulip-flutter/issues/613#issuecomment-2116318789) & [here](https://chat.zulip.org/#narrow/stream/48-mobile/topic/messages.20shift.20on.20mark-as-read/near/1811157).